examples/raspberrypi/rp2xxx/scripts/usb_device_loopback.py

```diff
@@ -10,39 +10,97 @@
 
 import usb.core
 import usb.util
+import time
 
-# find our device
-dev = usb.core.find(idVendor=0x0000, idProduct=0x0001)
-
-# was it found?
-if dev is None:
-    raise ValueError('Device not found')
-
-# get an endpoint instance
-cfg = dev.get_active_configuration()
-intf = cfg[(0, 0)]
-
-outep = usb.util.find_descriptor(
-    intf,
-    # match the first OUT endpoint
-    custom_match= \
-        lambda e: \
-            usb.util.endpoint_direction(e.bEndpointAddress) == \
-            usb.util.ENDPOINT_OUT)
-
-inep = usb.util.find_descriptor(
-    intf,
-    # match the first IN endpoint
-    custom_match= \
-        lambda e: \
-            usb.util.endpoint_direction(e.bEndpointAddress) == \
-            usb.util.ENDPOINT_IN)
-
-assert inep is not None
-assert outep is not None
-
-test_string = "Hello World!"
-outep.write(test_string)
-from_device = inep.read(len(test_string))
-
-print("Device Says: {}".format(''.join([chr(x) for x in from_device])))
+try:
+    # find our device
+    # use list_devices.py for idVendor and idProduct values
+    dev = usb.core.find(idVendor=0x0000, idProduct=0x0001)
+
+    # was it found?
+    if dev is None:
+        raise ValueError('Device not found. Verify that device is attached, then check vendor and product settings.')
+
+    print("Device found!")
+
+    # Detach kernel driver BEFORE setting configuration
+    detached_interfaces = []
+    for interface in [0, 1]:
+        if dev.is_kernel_driver_active(interface):
+            try:
+                dev.detach_kernel_driver(interface)
+                detached_interfaces.append(interface)
+                print(f"Kernel driver detached from interface {interface}")
+            except:
+                pass
+
+    # Set configuration
+    try:
+        dev.set_configuration()
+        print("Configuration set")
+    except usb.core.USBError as e:
+        print(f"Configuration already set or error: {e}")
+
+    # get an endpoint instance
+    cfg = dev.get_active_configuration()
+    intf = cfg[(1, 0)]  # Interface 1, Alt 0
+
+    outep = usb.util.find_descriptor(
+        intf,
+        custom_match=lambda e: usb.util.endpoint_direction(e.bEndpointAddress) == usb.util.ENDPOINT_OUT)
+
+    inep = usb.util.find_descriptor(
+        intf,
+        custom_match=lambda e: usb.util.endpoint_direction(e.bEndpointAddress) == usb.util.ENDPOINT_IN)
+
+    assert inep is not None, "IN endpoint not found"
```

Comment:
don't use asserts for this. Asserts are for when invariants in logic are not met. errors. You can just print an error and exit.